csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -249,6 +249,45 @@ class JumpReturnKind extends ReturnKind, TJumpReturnKind {
 
 class DataFlowCallable extends DotNet::Callable {
   DataFlowCallable() { this.isUnboundDeclaration() }
+
+  private string parameterQualifiedTypeNamesToString() {
+    result =
+      concat(Parameter p, int i |
+        p = this.getParameter(i)
+      |
+        p.getType().getQualifiedName(), "," order by i
+      )
+  }
+
+  /** Holds if a summary should apply for all overrides of this. */
+  predicate isBaseCallableOrPrototype() {
+    this.getDeclaringType() instanceof Interface
+    or
+    exists(Modifiable m | m = [this.(Modifiable), this.(Accessor).getDeclaration()] |
+      m.isAbstract()
+      or
+      this.getDeclaringType().(Modifiable).isAbstract() and m.(Virtualizable).isVirtual()
+    )
+  }
+
+  /** Gets a string representing whether a summary should apply for all overrides of this. */
+  private string getCallableOverride() {
+    if this.isBaseCallableOrPrototype() then result = "true" else result = "false"
+  }
+
+  /** Computes the first 6 columns for CSV rows. */
+  string asPartialModel() {
```

Comment:
This predicate and `isBaseCallableOrPrototype` will now be public (since `DataFlowCallable` is), which I don't think we want. Instead it would be better to add them as top-level predicates in `DataFlowPrivate.qll`.